packages/blocks/src/api/factory.js

```diff
@@ -289,7 +289,7 @@ const getBlockTypesForPossibleToTransforms = ( blocks ) => {
 
 	const sourceBlock = first( blocks );
 	const blockType = getBlockType( sourceBlock.name );
-	const transformsTo = getBlockTransforms( 'to', blockType.name );
+	const transformsTo = getBlockTransforms( 'to', blockType?.name );
```

Comment:
`getBlockTransforms` will return all block transforms when no name is specified, so I wonder if we should set this to an empty array instead for this case.

```jsx
const transformsTo = blockType ? getBlockTransforms( 'to', blockType.name ) : [];
```

Reply:
@gwwar how are testing it? I mean how do you come to know why `?.` operator won't work in this case.

Reply:
I read `getBlockTransforms` function to see what it does when `blockTypeOrName` is undefined. In this case it returns all block transforms. 

In the case that a block type is unknown or unregistered, it's probably more accurate that it's not allowed to transform into another block type rather than list all block types as a possible transformation. (Attempting to transform the block to anything else would likely error out).

https://github.com/WordPress/gutenberg/blob/7ae4f3d801e65cbc4a936ae0f45058d4392f6ba5/packages/blocks/src/api/factory.js#L397-L414